### DISCUSSION TITLE
only use emqx.conf can`t set node.name.

### DISCUSSION THREAD

**Discussion by panla** [emqx/emqx#8325] [Q&A] [answered]
### env

- emqx version: 4.3.15
- docker-compose

### desc

```text
if only use emqx.conf set node.name = emqx-1@172.20.7.3:
    got 7eeb45014d33@172.20.7.3
elif use emqx.conf and environment set EMQX_NODE_NAME=emqx-1@172.20.7.3:
    got emqx-1@172.20.7.3

```

**Comment by zmstone** [comment, marked as answer]
Hi @panla 
This is a known issue in docker entrypoint script.
Unfortunately we can’t change this behavior for compatibility reasons in 4.3 and 4.4.

5.0 has it changed to use `emqx` by default if neither of `EMQX_NAME` `EMQX_NODE_NAME` is not set.
